pymc3/distributions/bart.py

```diff
@@ -48,12 +51,24 @@ def __init__(self, X, Y, m=200, alpha=0.25, *args, **kwargs):
 
         self.num_observations = X.shape[0]
         self.num_variates = X.shape[1]
+        self.available_predictors = list(range(self.num_variates))
+        self.ssv = sample_splitting_variable(split_prior, self.num_variates)
         self.m = m
         self.alpha = alpha
         self.trees = self.init_list_of_trees()
+        self.all_trees = []
         self.mean = fast_mean()
         self.prior_prob_leaf_node = compute_prior_probability(alpha)
 
+    def preprocess_XY(self, X, Y):
+        if isinstance(Y, (Series, DataFrame)):
+            Y = Y.values
+        if isinstance(X, (Series, DataFrame)):
+            X = X.values
```

Comment:
quick comment - can we use `to_numpy` instead? https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.DataFrame.values.html?highlight=values#pandas.DataFrame.values

> We recommend using DataFrame.to_numpy() instead.

else this might happen:
```
>>> ser = pd.Series([1,2,3])
>>> ser = ser.astype('category')
>>> type(ser.values)
pandas.core.arrays.categorical.Categorical
```
which I guess isn't what's expected here?